File: frontend/openapi-generator.js
import { spawn } from 'child_process';
import fs from 'fs';

let outputDir = 'cmsTypeScriptClient';

// Function to delete directory recursively
function deleteDirectory(path) {
	if (fs.existsSync(path)) {
		fs.readdirSync(path).forEach((file) => {
			const curPath = path + '/' + file;
			if (fs.lstatSync(curPath).isDirectory()) {
				// Recurse
				deleteDirectory(curPath);
			} else {
				// Delete file
				fs.unlinkSync(curPath);
			}
		});
		fs.rmdirSync(path);
	}
}

deleteDirectory(outputDir);

// Run the openapi-generator-cli command

const command = 'C:/Program Files/nodejs/npx.cmd';
const args = [
	'@openapitools/openapi-generator-cli',
	'generate',
	'-i',
	'openapi.yaml',
	'-g',
	'typescript-fetch',
	'-o',
	outputDir,
	'-c',
	'openapi-generator.json',
	'--reserved-words-mappings',
	'delete=delete'
];

const generatorProcess = spawn(command, args);

generatorProcess.stdout.on('data', (data) => {
	console.log(`stdout: ${data}`);
});

generatorProcess.stderr.on('data', (data) => {
	console.error(`stderr: ${data}`);
});

generatorProcess.on('error', (error) => {
	console.error(`error: ${error.message}`);
});

generatorProcess.on('close', (code) => {
	console.log(`Command exited with code ${code}`);
});

File: frontend/openapiGenerator.java
import java.io.File;
import java.io.IOException;
import java.nio.file.Files;

public class openapiGenerator {

    public static void main(String[] args) {
        String outputDir = "cmsTypeScriptClient";

        // Function to delete directory recursively
        deleteDirectory(new File(outputDir));

        // Run the openapi-generator-cli command
        String command = "java";
        String[] commandArgs = {
                "-jar",
                "openapi-generator/modules/openapi-generator-cli/target/openapi-generator-cli.jar",
                "generate",
                "-i",
                "openapi.yaml",
                "-g",
                "typescript-fetch",
                "-o",
                outputDir,
                "-c",
                "openapi-generator.json",
                "--reserved-words-mappings",
                "delete=delete"
        };

        try {
            String[] fullCommand = new String[commandArgs.length + 1];
            fullCommand[0] = command;
            System.arraycopy(commandArgs, 0, fullCommand, 1, commandArgs.length);

            ProcessBuilder builder = new ProcessBuilder(fullCommand);
            Process process = builder.start();

            printOutput(process.getInputStream(), "stdout");
            printOutput(process.getErrorStream(), "stderr");

            int exitCode = process.waitFor();
            System.out.println("Command exited with code " + exitCode);
        } catch (IOException | InterruptedException e) {
            e.printStackTrace();
        }
    }

    private static void deleteDirectory(File directory) {
        if (directory.exists()) {
            File[] files = directory.listFiles();
            if (files != null) {
                for (File file : files) {
                    if (file.isDirectory()) {
                        // Recurse
                        deleteDirectory(file);
                    } else {
                        // Delete file
                        file.delete();
                    }
                }
            }
            directory.delete();
        }
    }

    private static void printOutput(java.io.InputStream inputStream, String type) throws IOException {
        java.io.BufferedReader reader = new java.io.BufferedReader(new java.io.InputStreamReader(inputStream));
        String line;
        while ((line = reader.readLine()) != null) {
            System.out.println(type + ": " + line);
        }
    }
}